testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -94,6 +111,10 @@ def test_check_set_unit_keyerror(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+    def test_check_set_unit_none_value(self, D):
+        units = {'density': None}
+        D._check_set_unit(units)
```

Comment:
Sorry it's late and I probably am missing what we're doing here but shouldn't we be checking that the units have been set properly?